packages/common/src/selling-partner-api-error.ts

```diff
@@ -0,0 +1,83 @@
+import {AxiosError} from 'axios'
+import {URL} from 'url'
+
+interface AmazonError {
+  messsage: string;
+  code: string;
+}
+
+interface Request {
+  pathname?: string;
+  domain?: string;
+  method?: string;
+  body?: string | Record<any, any>;
+  queries?: Record<any, any>;
+}
+
+interface Response {
+  errors: AmazonError[];
+  status: number;
+  headers: Record<string, string>;
+}
+
+export class SellingPartnerApiError extends Error {
+  public resource?: string
+  public action?: string
+  public response?: Response
+  public request?: Request
+  public apiName?: string
+  public apiVersion?: string
+
+  constructor(error: AxiosError) {
+    super()
+    let body = error.config.data
+    const url = error.config.url ? new URL(error.config.url) : null
+
+    try {
+      body = JSON.parse(body)
+    } catch {}
+
+    this.resource = error.config.url
+    this.action = error.config.method
```

Comment:
I’d rather name these with their actual names.

```suggestion
    this.url = error.config.url
    this.method = error.config.method
```